SIGSEGV in NameResolver.Resolve when `.await` is used as a property name

## Description

Type-aware linting crashes with a nil pointer dereference in `NameResolver.Resolve` when a file accesses `.await` as a property on an imported binding, in a file that also uses a JSDoc `@import` type annotation.

The crash is triggered by the `await` keyword being used as a property name via dot notation. The name resolver dereferences a nil scope/container for the `await` identifier node. Other keyword-as-property names (e.g., `.return`) do not trigger the crash.

## Motivation

This pattern is common in the [`@endo/patterns`](https://github.com/endojs/endo/tree/master/packages/patterns) library, where `M.await()` is a standard API for describing async method guard shapes:

```js
// from @endo/patterns patternMatchers.js
const M = freeze({
  // ...
  await: argPattern => makeAwaitArgGuard(argPattern),
  // ...
});
```

Consumer code like this is typical across the Agoric SDK (~29 call sites):

```js
M.interface('Foo', {
  getPublicFacet: M.callWhen(M.await(InstanceHandleShape)).returns(M.remotable()),
});
```

## Minimal reproduction (no third-party dependencies)

```js
import { join } from 'node:path';
/** @import {X} from 'y'; */
const x = join.await;
/** @type {X} */
let z;
```

Both conditions are required:
1. `.await` property access on an **imported** binding (local objects do not trigger the crash)
2. A JSDoc `@import` type that is **used** as a type annotation somewhere in the file

Removing either condition prevents the crash.

## Stack trace

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40]

github.com/microsoft/typescript-go/internal/binder.(*NameResolver).Resolve(...)
    typescript-go/internal/binder/nameresolver.go:96
github.com/microsoft/typescript-go/internal/checker.(*Checker).resolveEntityName(...)
    typescript-go/internal/checker/checker.go:15225
github.com/microsoft/typescript-go/internal/checker.(*Checker).getSymbolOfNameOrPropertyAccessExpression(...)
    typescript-go/internal/checker/checker.go:30830
github.com/microsoft/typescript-go/internal/checker.(*Checker).getSymbolAtLocation(...)
    typescript-go/internal/checker/checker.go:30595
github.com/microsoft/typescript-go/internal/checker.(*Checker).GetSymbolAtLocation(...)
    typescript-go/internal/checker/checker.go:30524
github.com/typescript-eslint/tsgolint/internal/rules/no_unnecessary_type_arguments.init.func1.1(...)
    no_unnecessary_type_arguments.go:33
```

Multiple type-aware rules trigger this through different call paths (`no_unnecessary_type_arguments`, `no_duplicate_type_constituents`, `no_redundant_type_constituents`) — all converge on the same nil pointer in `NameResolver.Resolve`.

## Possible upstream context

[microsoft/TypeScript#61144](https://github.com/microsoft/TypeScript/pull/61144) added `ignoreErrors=true` to `resolveEntityName` callers in the same call chain (`getSymbolOfNameOrPropertyAccessExpression`). This fix may not yet be ported to `typescript-go`.

## Workaround

Bracket notation avoids the crash because it produces a `StringLiteral` AST node instead of an `Identifier` node:

```js
// crashes
const x = M.await(shape);

// works
const x = M['await'](shape);
```

## Versions

- `oxlint`: 1.56.0
- `oxlint-tsgolint`: 0.17.0

--- a/patches/0007-fix-crash-on-JS-await-property-lookup.patch
+++ b/patches/0007-fix-crash-on-JS-await-property-lookup.patch
@@ -1,0 +1,28 @@
+From cbfcb4dddadb37c547b9c1f039625177716cf797 Mon Sep 17 00:00:00 2001
+From: Cameron Clark <cameron.clark@hey.com>
+Date: Thu, 19 Mar 2026 22:36:05 +0000
+Subject: [PATCH] fix crash on JS await property lookup
+
+---
+ internal/binder/nameresolver.go | 6 +++++-
+ 1 file changed, 5 insertions(+), 1 deletion(-)
+
+diff --git a/internal/binder/nameresolver.go b/internal/binder/nameresolver.go
+index b300126d09..eaf0d7ac35 100644
+--- a/internal/binder/nameresolver.go
++++ b/internal/binder/nameresolver.go
+@@ -93,7 +93,11 @@ loop:
+ 			}
+ 			fallthrough
+ 		case ast.KindModuleDeclaration:
+-			moduleExports := r.getSymbolOfDeclaration(location).Exports
++			moduleSymbol := r.getSymbolOfDeclaration(location)
++			if moduleSymbol == nil {
++				break
++			}
++			moduleExports := moduleSymbol.Exports
+ 			if ast.IsSourceFile(location) || (ast.IsModuleDeclaration(location) && location.Flags&ast.NodeFlagsAmbient != 0 && !ast.IsGlobalScopeAugmentation(location)) {
+ 				// It's an external module. First see if the module has an export default and if the local
+ 				// name of that export default matches.
+-- 
+2.50.1 (Apple Git-155)

